pkg/performanceprofile/controller/performanceprofile/components/manifestset/manifestset.go

```diff
@@ -53,7 +56,13 @@ func (ms *ManifestResultSet) ToManifestTable() ManifestTable {
 func GetNewComponents(profile *performancev2.PerformanceProfile, opts *components.Options) (*ManifestResultSet, error) {
 	machineConfigPoolSelector := profilecomponent.GetMachineConfigPoolSelector(profile, opts.ProfileMCP)
 
-	mc, err := machineconfig.New(profile, &opts.MachineConfig)
+	autogen := profileutil.HaveEnablementAutogeneration(profile)
+	llcEnabled := profileutil.IsLLCAlignmentEnabled(profile)
+	klog.V(4).InfoS("components manifests", "autogenerateEnablement", autogen, "LLCEnabled", llcEnabled)
+
+	mcOpts := opts.MachineConfig.Clone()
+	mcOpts.LLCFileEnabled = autogen && llcEnabled
```

Comment:
What we need is also - when autogen == false force llcEnabled to false.
Then not generating the file is cleaner, but makes no functional difference to kubelet.

Reply:
ok, this means we need a different feature then. Is no longer a simplification but rather a master switch. At this point we should evaluate if we should add to our APIs and take full ownership of the flag.